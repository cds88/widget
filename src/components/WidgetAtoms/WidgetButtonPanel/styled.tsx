import Box from '@mui/material/Box';
import { styled } from '@mui/system';

export const WidgetButtonPanelWrapperBox = styled(Box)`
  flex: 1;
  align-items: center;
  cursor: pointer;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: '2px 0';
  color:linear-gradient(white, #f0f0f0);
`;
