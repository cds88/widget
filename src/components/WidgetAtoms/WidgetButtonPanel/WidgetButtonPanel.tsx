import {  BoxProps, Switch, Typography } from '@mui/material';
import React, { ChangeEvent } from 'react';
import { WidgetButtonPanelWrapperBox } from './styled';


type WidgetButtonPanelProps = {
  children: React.ReactNode;
  checked: boolean;
  onChange: (value: boolean) => void;
} & Omit<BoxProps, 'onChange' | 'sx'>

const WidgetButtonPanel: React.FC<WidgetButtonPanelProps> = ({
  children,
  checked,
  onChange,
  ...props
}) => {
  const handleToggle = (event: ChangeEvent<HTMLInputElement>) => {
    onChange(event.target.checked);
  };

  return (
    <WidgetButtonPanelWrapperBox {...props}>
      <Typography variant="h4">{children}</Typography>
      <Switch checked={checked} onChange={handleToggle} />
    </WidgetButtonPanelWrapperBox>
  );
};

export default WidgetButtonPanel;
