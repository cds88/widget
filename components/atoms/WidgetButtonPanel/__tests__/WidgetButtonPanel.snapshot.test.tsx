import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import WidgetButtonPanel from '../WidgetButtonPanel';

describe('WidgetButtonPanel', () => {
  const mockOnChange = jest.fn();

  beforeEach(() => {
    mockOnChange.mockClear();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <WidgetButtonPanel checked={false} onChange={mockOnChange}>
        Test Widget
      </WidgetButtonPanel>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
