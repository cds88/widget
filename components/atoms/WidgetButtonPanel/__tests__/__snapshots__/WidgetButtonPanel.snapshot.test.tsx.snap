// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetButtonPanel matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-9c42oi"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-10zsxzp-MuiTypography-root"
    >
      Test Widget
    </h4>
    <span
      class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
    >
      <span
        class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
      >
        <input
          class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
          type="checkbox"
        />
        <span
          class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
        />
      </span>
      <span
        class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
      />
    </span>
  </div>
</DocumentFragment>
`;
